### ISSUE TITLE
Processing of “chunked” Base64 encoded (typically Malware) parts

### ISSUE THREAD

**Issue by albrechtd** [jstedfast/gmime#150]
While re-building some of my software for Debian Bookworm I stumbled over different decoding of “chunked” Base64 encoded parts which are sometime used by Malware authors (see [here](https://noxxi.de/research/mime-5-easy-steps-to-bypass-av.html#hdr1.5) for a description):

On Debian Bullseye, coming with GMime 3.2.7, the output of the [attached trivial test application](https://github.com/jstedfast/gmime/files/11480905/test-gmime-base64.c.gz) is  
```
** Message: 20:25:26.437: body: 'VIRUS!!'
```
i.e. GMime does *not* stop decoding at the very first “=”.

However, on Debian Bookworm, with GMime 3.2.13, this is not the case, as the result reads
```
** Message: 20:25:27.947: body: 'VI'
```

As my application tries to detect anomalies in messages *and* to completely extract possibly malicious content for further inspection, the new GMime version actually breaks this feature.  I would prefer to avoid writing my own decoder – would it be possible to re-enable decoding chunked Base64 completely?  This should make absolutely no difference for any well-formed messages, but would be extremely helpful for use cases like mine.

Thanks, Albrecht.

**Comment by jstedfast** [comment]
Oops, reverted the wrong thing.

**Comment by jstedfast** [comment]
I'll have to look at this again next time I'm on my Mac. Currently on Windows so I can't build or run the unit tests :(

**Comment by jstedfast** [comment]
Edited the code on Windows so I'm not even sure the above commit will compile, but theoretically, at least, it should be what is needed for this.

**Comment by albrechtd** [comment]
> Edited the code on Windows so I'm not even sure the above commit will compile, but theoretically, at least, it should be what is needed for this.

The code compiles on _Bookworm_, with a (probably unrelated) warning:
```
url-scanner.c: In function 'url_web_end':
url-scanner.c:336:21: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  336 |                 end = inptr;
      |                     ^
```
and works as expected.  Thanks for fixing!

**Comment by jstedfast** [comment]
I think that is related to a previous change I made to fix the url-scanner. I'll have to take a look at that.

**Comment by jstedfast** [comment]
just released 3.2.14 with this fix
